### DISCUSSION TITLE
Non-present languages in my list

### DISCUSSION THREAD

**Discussion by Tachytaenius** [anuraghazra/github-readme-stats#3501] [Q&A] [answered]
### Describe the bug

When I preview changes (showing more languages and excluding some more repos) to my language counter on my profile, it's convinced that there are tiny trace amounts of PHP and Pawn, which I have never programmed in before, and C++ which I have no non-forked public repositories of (though if I do a search for C++ files under my user that are not in forks, I get a bunch of assembly-related files (but PHP and Pawn do come up totally empty)).

### Expected behavior

PHP and Pawn should not be present, and neither should C++, though I think that is on GitHub's end if it comes up with files that aren't C++ in a C++ search?

### Screenshots / Live demo link

![](https://github-readme-stats.vercel.app/api/top-langs/?username=Tachytaenius&langs_count=12&layout=compact&theme=omni&hide_border=true&border_radius=10&exclude_repo=pokestygian,pokehalls,Better-Mushroom-Islands,Magical-Mages,ChocolateArmoury)

**Comment by qwerty541** [comment, marked as answer]
Hey, @Tachytaenius! We use GitHub GraphQL API under the hood to fetch stats data. It show that this repository contains PHP: https://github.com/Tachytaenius/game-boy-n-body. You can personally check the data which we receive from GitHub by running the following query in [GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer):

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    # fetch only owner repos & not forks
    repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

```
{ "login":"Tachytaenius" }
```

If it count some other files as PHP by mistake you can contact GitHub support about that. Our team can't do anything, correctness of https://github.com/github-linguist/linguist is not our responsibility. I can only suggest you to try silence these files using configuration of `.gitattributes` file.

**Reply by rickstaa** [reply]
In addition to @qwerty541's response, there's an ongoing need to enhance the language algorithm (refer to the GitHub issue [here](https://github.com/anuraghazra/github-readme-stats/issues/1852)). It retrieves only the initial 100 repositories, leading to inaccuracies for users with an extensive repository count. A pull request has been initiated to address this issue, and upon approval, we'll release it as an environment variable, enabling usage within [GitHub actions](https://github.com/anuraghazra/github-readme-stats/issues/2179).

Apart from that, due to GraphQL limitations, we are currently constrained to fetching language data on a repository level rather than a user basis. I've initiated [a feature request](https://github.com/orgs/community/discussions/18230) with GitHub to enhance this behaviour. Your support for this feature request is crucial in drawing attention to the issue at GitHub and increasing the likelihood of a solution.
